frontend/src/components/chat-interface.tsx

```diff
@@ -125,6 +128,23 @@ export function ChatInterface() {
             </ChatMessage>
           ),
         )}
+
+        {messages.length > 2 &&
+          !isErrorMessage(lastMessage) &&
+          lastMessage.sender === "assistant" && (
```

Comment:
I expected this to trigger if AgentState flips to WAITING_FOR_USER_INPUT or FINISHED. Based on the current logic, I think it'd show even if the agent is in the middle of working, so long as it printed a message

Reply:
Good point!